uberfire-commons/src/main/java/org/uberfire/commons/cluster/ClusterJMSService.java

```diff
@@ -107,28 +107,44 @@ private ClusterParameters loadParameters() {
                                    String channel,
                                    Class<T> objectMessageClass,
                                    Consumer<T> listener) {
-        try {
-            Session session = createConsumerSession();
-            Destination topic = createDestination(type,
-                                                  channel,
-                                                  session);
-            MessageConsumer messageConsumer = session.createConsumer(topic);
-
-            messageConsumer.setMessageListener(message -> {
-                if (message instanceof ObjectMessage) {
-                    try {
-                        Serializable object = ((ObjectMessage) message).getObject();
-                        if (objectMessageClass.isInstance(object)) {
-                            listener.accept((T) object);
+
+        consumerSessions.compute(channel, (s, session) -> {
```

Comment:
computeIfAbsent

Reply:
But I need to check that null in case the session couldn't be created, compute is also atomic as computIfAbsent